recipes/uriparser/build.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -ex
+
+mkdir -p build-cpp
+pushd build-cpp
+
+cmake -GNinja \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_PREFIX_PATH=$PREFIX \
+      -DCMAKE_INSTALL_PREFIX=$PREFIX \
+      -DCMAKE_C_FLAGS="$CFLAGS" \
+      -DCMAKE_POSITION_INDEPENDENT_CODE=on \
+      -DURIPARSER_BUILD_DOCS=off \
+      -DURIPARSER_BUILD_TESTS=off \
+      -DURIPARSER_BUILD_TOOLS=off \
+      -DURIPARSER_BUILD_WCHAR_T=off \
+      -DBUILD_SHARED_LIBS=off \
```

Comment:
Shared libs would be preferred

Reply:
Not sure that would work with Arrow...

Reply:
Well, you can build & install both the shared and static libs.  Some other recipes do this, as it provides the highest level of downstream support.

Reply:
Arrow should work with the shared lib, otherwise I'll fix that.

@scopatz We need to figure out something with static libs and also headers in future in conda-forge (maybe `-dev` packages) as they amount to quite some decent size increases when building shareables like docker containers. I'll raise an issue on https://github.com/conda-forge/conda-forge.github.io/issues once I have the time to come up with a good proposal.

Reply:
@scopatz I'm afraid the `.lib` file would conflict on Windows.